app/assets/javascripts/map.js

```diff
@@ -80,39 +100,101 @@ class Map {
       componentRestrictions: {country: 'ca'}
     };
 
-    const input = document.getElementById('address');
+    const input = document.getElementById('addressInput');
     const autocomplete = new google.maps.places.Autocomplete(input, options);
+    // TODO execute search when autocomplete option has been selected doesnt work
+    // autocomplete.addListener('place_changed', this.handleSearchSubmit);
   }
 
   addListeners() {
-    document.getElementById('AddressSearchModal').addEventListener('submit', this.handleSearchSubmit);
+    document.getElementById('addressSearchBar').addEventListener('keyup', this.enableFindButton)
+    document.getElementById('addressSearchBar').addEventListener('submit', this.handleSearchSubmit);
     this.drawingManager.addListener('polygoncomplete', this.handlePolygonCreated);
     this.removeLastControl.addEventListener('click', this.handleRemoveLastPolygon);
     this.removeAllControl.addEventListener('click', this.handleRemoveAllPolygons);
+    document.getElementById('doneSelectingArea').addEventListener('click', this.handleDoneSelecting);
+    document.getElementsByName("serviceExpeditionCost").forEach((element) => { 
+      element.addEventListener('click', this.handleExpeditionInfoClick); 
+    });
```

Comment:
This is normally handled better with a [single event listener on the parent](https://davidwalsh.name/event-delegate) (instead of N listeners). It's especially important when you have a large number of child nodes (not the case here, but just FYI).

This is called Event Delegation